BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/SelfDiagnostics/SelfDiagnosticsConfigParser.cs

```diff
@@ -31,11 +36,149 @@ internal class SelfDiagnosticsConfigParser
         // In theory the variable won't be access at the same time because worker thread first Task.Delay for a few seconds.
         private byte[] configBuffer;
 
+        /// <summary>
+        /// Represents the location for App Insights to parse user-defined self-diagnostics settings.
+        /// </summary>
+        internal enum ParseLocation
+        {
+            /// <summary>
+            /// Parse self-diagnostics settings from enviornment variable(s).
+            /// </summary>
+            EnviornmentVariable,
+
+            /// <summary>
+            /// Parse self-diagnostics settings from tje JSON file.
+            /// </summary>
+            ConfigJson,
+        }
+
         public bool TryGetConfiguration(out string logDirectory, out int fileSizeInKB, out EventLevel logLevel)
+        {
+            if (TryGetConfigFromEnvrionmentVariable(out logDirectory, out fileSizeInKB, out logLevel))
```

Comment:
Could this method ever throw exception? What would happen if it does?

Reply:
The only place I can see now exception might happen is here:

``
logLevel = (EventLevel)Enum.Parse(typeof(EventLevel), val);
``

Shall we do try/catch in the scope of `TryGetConfigFromEnvrionmentVariable` to accommodate the exceptions that we can't foresee, or try/catch the above mentioning Enum.Parse part should be sufficient?

Reply:
By the way, `Environment.GetEnvironmentVariable` could also throw exceptions.

Reply:
I've added try/catch to `TryGetConfigFromEnvrionmentVariable`.
Currently, unlike in `TryGetConfigFromJsonFile`, I use `throw` when there is an exception because I think it would be nice to get users attention when configs set via environment variables are not being parsed correctly.

Reply:
Be careful of in what scenarios this happens. It could be in production code. For the local file configuration approach, if the user accidentally has a bad config, their app will crash because of it. I'm not sure about the environment variable.

The question boils down to: "can the app be crashed while running because of user/admin modified an environment variable?"

Reply:
Ah I see what you mean! 
Maybe we can write some logs for this case instead of throwing.
I'm just not sure where to write the exception logs...

Reply:
Hey team, one question I have.... can we toggle environment variable values of a running process? Say customer has a production app, I want to simply start collecting self-diagnostic logs, then find some way to set that environment variable for the host environment, will the tracing start logging within 10 seconds? Then when I want to stop that tracing, can I just adjust the env var and see tracing stop within 10 seconds? This behavior works when we rely on reading these settings from Json file, but I'm not certain if a "refresh" of environment variables requires a process restart each time? My original goal was to have env var that deteremines what path the SDK will poll every 10 seconds for the json file with those settings. Does this direct environment variable settings, where you logDirectory, file size, loglevel still give us an opportunity to turn on/off tracing of this production running app without restarts?

Reply:
I will collaborate with Rohit in testing the above scenarios and update here.